### PR TITLE
[7.x] Custom component prefix

### PR DESCRIPTION
**Pull request by driesvints**
Set a custom prefix for components.

```php
Blade::prefix('z');
```

### CODE REVIEW

File: src/Illuminate/Support/Facades/Blade.php [hunk 1]
```diff
@@ -13,6 +13,7 @@
  * @method static void if(string $name, callable $callback)
  * @method static bool check(string $name, array ...$parameters)
  * @method static void component(string $path, string|null $alias = null)
+ * @method static void prefix(string $prefix)
```
Comment (GrahamCampbell) [inline src/Illuminate/Support/Facades/Blade.php]: Really not obvious this is for components from the name. What about `componentPrefix`?

**Review by GrahamCampbell — CHANGES_REQUESTED** [review]

File: src/Illuminate/Support/Facades/Blade.php (hunk 1 above)
Comment (driesvints) [reply src/Illuminate/Support/Facades/Blade.php]: Maybe yeah. We don't use prefixes anywhere else though?

**Comment by mpociot** [comment]
I would prefer „componentPrefix“ as well. Just to make it clear what this is for.

**Comment by driesvints** [comment]
Okay renamed it.

**Comment by taylorotwell** [comment]
Personally I think I will hold off on this. I view it as a similar thing as `app:name`. Yes, it provides customization but at the cost of making switching between projects and standardization difficult. I don't see a big need to ever do this so until someone has a very good reason I'm going to wait.

**Comment by driesvints** [comment]
Sure 👍

**Pull request closed without merging**